src/main/java/teammates/common/datatransfer/FeedbackQuestionAttributes.java

```diff
@@ -528,15 +540,15 @@ public void setQuestionDetails(FeedbackQuestionDetails questionDetails) {
      * @return The Feedback*QuestionDetails object representing the question's details
      */
     public FeedbackQuestionDetails getQuestionDetails() {
-        // For Text questions, the questionText simply contains the question, not a JSON
+        // For old Text questions, the questionText simply contains the question, not a JSON
         // This is due to legacy data in the data store before there are multiple question types
-        if (questionType == FeedbackQuestionType.TEXT) {
+        if (questionType == FeedbackQuestionType.TEXT && !isJSONValid(questionMetaData.getValue())) {
             return new FeedbackTextQuestionDetails(questionMetaData.getValue());
```

Comment:
@kanghj For handling old Text Questions, that only have Text and are not is JSON string. Is that ok ?